accurest-core/src/main/groovy/io/codearte/accurest/builder/SpockMethodBodyBuilder.groovy

```diff
@@ -61,12 +71,24 @@ class SpockMethodBodyBuilder {
 			if (response.body) {
 				endBlock()
 				addLine('and:').startBlock()
-				addLine('def responseBody = new JsonSlurper().parseText(response.body.asString())')
 				def responseBody = response.body.serverValue
-				if (responseBody instanceof List) {
-					processArrayElements(responseBody, "", blockBuilder)
-				} else {
-					processMapElement(responseBody, blockBuilder, "")
+				ContentType contentType = recognizeContentTypeFromHeader(response.headers)
+				if (contentType == ContentType.UNKNOWN) {
```

Comment:
this whole part can be extracted to some abstraction.

Also you can use `switch` on  type with Groovy - http://docs.groovy-lang.org/latest/html/documentation/#_switch_case